server_test.go

```diff
@@ -1944,6 +1964,7 @@ func TestServerProcessOutboundQos2Flow(t *testing.T) {
 	for i, tx := range tt {
 		t.Run("qos step"+strconv.Itoa(i), func(t *testing.T) {
 			r, w := net.Pipe()
+			time.Sleep(time.Millisecond)
```

Comment:
Had to add a few delays as there is a slight latency between adding a message to the queue and it being delivered